### ISSUE TITLE
[BUG] make failed on v0.2 branch

### ISSUE THREAD

**Issue by avairavan** [harvester/harvester#721]
**Describe the bug**
I checked out the branch origin/v0.2 from the harvester git repo and built it using "make" command. The build failed with the following details:

**To Reproduce**
git clone https://github.com/rancher/harvester.git
git checkout origin/v0.2
cd harvester
make

**Expected behavior**
Compiled harvester docker image.

**Log**
Step 9/13 : RUN mkdir -p /usr/share/rancher/harvester &&     cd /usr/share/rancher/harvester &&     curl -sL https://releases.rancher.com/harvester-ui/${HARVESTER_UI_VERSION}.tar.gz | tar xvzf - --strip-components=2 &&     mkdir -p /usr/share/rancher/harvester/api-ui &&     cd /usr/share/rancher/harvester/api-ui &&     curl -sL https://releases.rancher.com/api-ui/${HARVESTER_API_UI_VERSION}.tar.gz | tar xvzf - --strip-components=1 &&     cd /var/lib/rancher/harvester
 ---> Running in a4180327320b
gzip: invalid magic
tar: Child returned status 1
tar: Error is not recoverable: exiting now
The command '/bin/sh -c mkdir -p /usr/share/rancher/harvester &&     cd /usr/share/rancher/harvester &&     curl -sL https://releases.rancher.com/harvester-ui/${HARVESTER_UI_VERSION}.tar.gz | tar xvzf - --strip-components=2 &&     mkdir -p /usr/share/rancher/harvester/api-ui &&     cd /usr/share/rancher/harvester/api-ui &&     curl -sL https://releases.rancher.com/api-ui/${HARVESTER_API_UI_VERSION}.tar.gz | tar xvzf - --strip-components=1 &&     cd /var/lib/rancher/harvester' returned a non-zero code: 2
FATA[1166] exit status 2
make: *** [Makefile:11: default] Error 1

**Environment:**
 - Harvester ISO version: v0.2
 - Installation Mode: app-mode
 - Underlying Infrastructure Baremetal

**Additional context**

**Comment by gitlawr** [comment]
The build command failed. It could be a network issue. 
Can you check if you can fetch the UI file in your environment?
```
wget https://releases.rancher.com/harvester-ui/v0.2.0-rc3.tar.gz
```

**Comment by avairavan** [comment]
Yes. I can fetch it manually. But I think it fails when done within the alpine container.

`/tmp# wget https://releases.rancher.com/harvester-ui/v0.2.0-rc3.tar.gz
--2021-04-20 06:38:47--  https://releases.rancher.com/harvester-ui/v0.2.0-rc3.tar.gz
Resolving releases.rancher.com (releases.rancher.com)... 104.26.12.240, 104.26.13.240, 172.67.69.133, ...
Connecting to releases.rancher.com (releases.rancher.com)|104.26.12.240|:443... connected.
HTTP request sent, awaiting response... 200 OK
Length: 9315510 (8.9M) [application/gzip]
Saving to: ‘v0.2.0-rc3.tar.gz’

v0.2.0-rc3.tar.gz                 100%[============================================================>]   8.88M  12.5MB/s    in 0.7s

2021-04-20 06:38:48 (12.5 MB/s) - ‘v0.2.0-rc3.tar.gz’ saved [9315510/9315510]

`

**Comment by guangbochen** [comment]
optionally u can use an EC2 instance to make the build as it will be efficient, otherwise u can replace the assets url at https://github.com/rancher/harvester/blob/master/package/Dockerfile#L18 with a custom internal url as a workaround.

**Comment by avairavan** [comment]
Thanks. Can you please explain how building it in the EC2 instance will solve the problem?

**Comment by gitlawr** [comment]
1. add `set -x;` before the command in Step 9/13 then do the make to see which subcommand explicitly fails
2. run the commands in alpine for debugging.
e.g.,
```
docker run -it alpine sh

# run in the alpine container
apk update && apk add -u --no-cache git curl unzip tar tini bash nfs-utils && \
    mkdir -p /var/lib/rancher/harvester 
set -x;
mkdir -p /usr/share/rancher/harvester && \
    cd /usr/share/rancher/harvester && \
    curl -sL https://releases.rancher.com/harvester-ui/v0.2.0-rc3.tar.gz | tar xvzf - --strip-components=2 && \
    mkdir -p /usr/share/rancher/harvester/api-ui && \
    cd /usr/share/rancher/harvester/api-ui && \
    curl -sL https://releases.rancher.com/api-ui/1.1.9.tar.gz | tar xvzf - --strip-components=1 && \
    cd /var/lib/rancher/harvester
```

**Comment by avairava** [comment]
I circumvented the issue by downloading the tar files separately using wget and modified the Dockerfile as shown below:

```
wget https://releases.rancher.com/harvester-ui/${HARVESTER_UI_VERSION}.tar.gz
wget https://releases.rancher.com/api-ui/${HARVESTER_API_UI_VERSION}.tar.gz 
```

```
`diff --git a/package/Dockerfile b/package/Dockerfile
index 6053cab2..59010854 100644
--- a/package/Dockerfile
+++ b/package/Dockerfile
@@ -14,12 +14,13 @@ ENV HARVESTER_API_UI_VERSION 1.1.9
 ARG VERSION=dev
 ENV HARVESTER_SERVER_VERSION ${VERSION}
 RUN mkdir -p /usr/share/rancher/harvester && \
-    cd /usr/share/rancher/harvester && \
-    curl -sL https://releases.rancher.com/harvester-ui/${HARVESTER_UI_VERSION}.tar.gz | tar xvzf - --strip-components=2 && \
-    mkdir -p /usr/share/rancher/harvester/api-ui && \
-    cd /usr/share/rancher/harvester/api-ui && \
-    curl -sL https://releases.rancher.com/api-ui/${HARVESTER_API_UI_VERSION}.tar.gz | tar xvzf - --strip-components=1 && \
-    cd /var/lib/rancher/harvester
+    mkdir -p /usr/share/rancher/harvester/api-ui
+
+COPY ${HARVESTER_UI_VERSION}.tar.gz /usr/share/rancher/harvester/
+COPY ${HARVESTER_API_UI_VERSION}.tar.gz /usr/share/rancher/harvester/api-ui
+RUN cd /usr/share/rancher/harvester && tar xvzf ${HARVESTER_UI_VERSION}.tar.gz --strip-components=2
+RUN cd /usr/share/rancher/harvester/api-ui && tar xvzf ${HARVESTER_API_UI_VERSION}.tar.gz --strip-components=1
+RUN cd /var/lib/rancher/harvester

 COPY entrypoint.sh harvester /usr/bin/
 RUN chmod +x /usr/bin/entrypoint.sh
`
```
